hollow/src/main/java/com/netflix/hollow/api/producer/validation/DuplicateDataDetectionValidator.java

```diff
@@ -64,17 +65,33 @@ public DuplicateDataDetectionValidator(String dataTypeName, String[] fieldPathNa
 	 */
 	@Override
 	public void validate(ReadState readState) {
-		log.log(Level.INFO, "Running DuplicateDataDetectionValidator for type "+dataTypeName);
-		PrimaryKey primaryKey = getPrimaryKey(readState);
-		HollowPrimaryKeyIndex hollowPrimaryKeyIndex = new HollowPrimaryKeyIndex(readState.getStateEngine(), primaryKey);
+		Builder statusBuilder = initializeForValidation(readState);
+		
+		PrimaryKey primaryKey = getPrimaryKey(readState, statusBuilder);
+		String fieldPaths = Arrays.toString(primaryKey.getFieldPaths());
+		statusBuilder.addAdditionalInfo(FIELD_PATH_NAME , fieldPaths);
+		
+		HollowPrimaryKeyIndex hollowPrimaryKeyIndex = new HollowPrimaryKeyIndex(readState.getStateEngine(), primaryKey );
```

Comment:
Wonder if it should try to use existing before creating new one for example: HollowPrimaryKeyIndex primaryKeyIndex = typeState.getListener(HollowPrimaryKeyIndex.class);

Reply:
Based on what I remember that is what the call does underneath. I will double check.

Reply:
I remembered wrong. Modified to reused if one is existing.